### PR TITLE
[numpy] operator ravel, derive from reshape

### PR DESCRIPTION
**Pull request by tingying2020**
Numpy operator ravel which is the same as reshape(x, -1).

@haojin2

### CODE REVIEW

File: tests/python/unittest/test_numpy_op.py [hunk 1]
```diff
@@ -1082,6 +1082,45 @@ def hybrid_forward(self, F, a, *args):
...
+    types = ['float64', 'float32', 'float16', 'int64', 'int32', 'int8']
+    for oneType in types:
+        for hybridize in [True, False]:
+            for shape in [(),
+                          (2,),
+                          (2, 2),
+                          (1, 2, 3),
+                          (3, 0),
+                          (1, 0, 2)
+                          ]:
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: Simply do
```Python
for shape in [(), (2,), (2, 2), (1, 2, 3), (3, 0), (1, 0, 2)]:
```

Comment (tingying2020) [reply tests/python/unittest/test_numpy_op.py]: Done. Thanks.

File: python/mxnet/ndarray/numpy/_op.py [hunk 2]
```diff
@@ -2363,3 +2363,55 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):  # pylint:
...
+    made only if needed.
+
+    Parameters
+    ----------
+    x : ndarray
+        Input array.  The elements in `x` are read in row-major, C-style order and
+        packed as a 1-D array.
+    out : ndarray or None, optional
+        A location into which the result is stored. If not provided or `None`,
+        a freshly-allocated array is returned.
```
Comment (sxjscience) [inline python/mxnet/ndarray/numpy/_op.py]: I find that out is actually not used.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 3]
```diff
@@ -2678,4 +2678,42 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):  # pylint:
...
+        Note that matrices are special cased for backward compatibility, if `x`
+        is a matrix, then y is a 1-D ndarray.
+
+    Notes
+    -----
+    This function differs from the original numpy.arange in the following aspects:
+        - Only support row-major, C-style order.
+    """
+    if isinstance(x, numeric_types):
+        return _np.reshape(x, -1)
```
Comment (sxjscience) [inline python/mxnet/symbol/numpy/_symbol.py]: We need to set the `out` when it's not None.

File: tests/python/unittest/test_numpy_op.py [hunk 4]
```diff
@@ -1323,6 +1323,39 @@ def hybrid_forward(self, F, a, *args):
...
+            for shape in [(), (2,), (2, 2), (1, 2, 3), (3, 0), (1, 0, 2)]:
+                test_ravel = TestRavel()
+                if hybridize:
+                    test_ravel.hybridize()
+                x = rand_ndarray(shape, dtype=oneType).as_np_ndarray()
+                x.attach_grad()
+                np_out = _np.ravel(x.asnumpy())
+                with mx.autograd.record():
+                    mx_out = test_ravel(x)
+                assert mx_out.shape == np_out.shape
```
Comment (sxjscience) [inline tests/python/unittest/test_numpy_op.py]: Would you also add test for the return type of the output?

File: tests/python/unittest/test_numpy_op.py [hunk 5]
```diff
@@ -1744,6 +1777,45 @@ def test_indexing_mode(sampler, set_size, samples_size, replace, weight=None):
...
+                          ]:
+                test_ravel = TestRavel()
+                if hybridize:
+                    test_ravel.hybridize()
+                x = rand_ndarray(shape, dtype=oneType).as_np_ndarray()
+                x.attach_grad()
+                np_out = _np.ravel(x.asnumpy())
+                with mx.autograd.record():
+                    mx_out = test_ravel(x)
+                assert mx_out.shape == np_out.shape
```
Comment (sxjscience) [inline tests/python/unittest/test_numpy_op.py]: Also test the data type here.

File: python/mxnet/ndarray/numpy/_op.py [hunk 6]
```diff
@@ -2363,3 +2363,55 @@ def var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):  # pylint:
     0.2025
     """
     return _npi.var(a, axis=axis, dtype=dtype, ddof=ddof, keepdims=keepdims, out=out)
+
+
+@set_module('mxnet.ndarray.numpy')
+def ravel(x, out=None):
```
Comment (sxjscience) [inline python/mxnet/ndarray/numpy/_op.py]: In numpy, we have the `order='C'` argument. We need to add that and raise NotImplementedError if the value is other than `C`.

File: python/mxnet/symbol/numpy/_symbol.py (hunk 3 above)
Comment (sxjscience) [reply python/mxnet/symbol/numpy/_symbol.py]: Actually, I find that there is no `out` argument in numpy.

File: python/mxnet/ndarray/numpy/_op.py (hunk 2 above)
Comment (sxjscience) [reply python/mxnet/ndarray/numpy/_op.py]: I checked the numpy doc, there is no out argument.

**Comment by sxjscience** [comment]
@tingying2020 Would you make a small change in the code to trigger the CI again?

**Comment by tingying2020** [comment]
> @tingying2020 Would you rebase the code? I will merge after that.

**Pull request merged**